### PR TITLE
config: add -c, --config commandline options

### PR DESCRIPTION
**Pull request by bvr-yr**
just copy-pasted from hypridle
strange it's not here already

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -24,6 +26,9 @@ int main(int argc, char** argv, char** envp) {
         else if (arg == "--quiet" || arg == "-q")
             Debug::quiet = true;
 
+        else if (arg == "--config" || arg == "-c")
```
Comment (vaxerski) [inline src/main.cpp:29]: missing check for argc

Comment (bvr-yr) [reply src/main.cpp]: done

File: src/config/ConfigManager.cpp [hunk 2]
```diff
@@ -30,8 +30,8 @@ static std::string getMainConfigPath() {
     return getConfigDir() + "/hypr/hyprlock.conf";
 }
 
-CConfigManager::CConfigManager() : m_config(getMainConfigPath().c_str(), Hyprlang::SConfigOptions{.throwAllErrors = true, .allowMissingConfig = true}) {
-    configCurrentPath = getMainConfigPath();
+CConfigManager::CConfigManager(std::string configPath) : m_config(configPath.empty() ? getMainConfigPath().c_str() : configPath.c_str(), Hyprlang::SConfigOptions{.throwAllErrors = true, .allowMissingConfig = true}) {
```
Comment (vaxerski) [inline src/config/ConfigManager.cpp:33]: you shouldn't allow a missing config in the case of a custom config path

Comment (bvr-yr) [reply src/config/ConfigManager.cpp]: done

**Review by vaxerski — APPROVED** [review]
lgtm

**Pull request merged**